path_metadata.go

```diff
@@ -136,11 +145,69 @@ func (b *versionedKVBackend) pathMetadataRead() framework.OperationFunc {
 				"max_versions":         meta.MaxVersions,
 				"cas_required":         meta.CasRequired,
 				"delete_version_after": deleteVersionAfter.String(),
+				"custom_metadata":      meta.CustomMetadata,
 			},
 		}, nil
 	}
 }
 
+const maxCustomMetadataKeys = 64
+const maxCustomMetadataKeyLength = 128
+const maxCustomMetadataValueLength = 512
+const customMetadataValidationErrorPrefix = "custom_metadata validation failed"
+
+// Perform input validation on custom_metadata field. If the key count
+// exceeds maxCustomMetadataKeys, the validation will be short-circuited
+// to prevent unnecessary (and potentially costly) validation to be run.
+// If the key count falls at or below maxCustomMetadataKeys, multiple
+// checks will be made per key and value. These checks include:
+//   - 0 < length of key <= maxCustomMetadataKeyLength
+//   - 0 < length of value <= maxCustomMetadataValueLength
+//   - keys and values cannot include unprintable characters
+func validateCustomMetadata(customMetadata map[string]string) error {
+	var errs *multierror.Error
+
+	if keyCount := len(customMetadata); keyCount > maxCustomMetadataKeys {
```

Comment:
If you see much of Vault's code, it leans towards returning early to keep the code _relatively_ easy to read and comprehend. What I mean to say is that the else block here avoided and bring the code within it outside to make it more flat.

if condition failed {
  return err
}

// The code that went in the else block can be written here without the surrounding braces.
// The multi error declaration can be done at this point.

WDYT?

Reply:
An early return here would be much cleaner. I think, however, that I'll still use a `multierror` in both cases. Doing so will provide consistency in the structure of the JSON response body.

Reply:
Will it differ if the error is an `error` vs `multierror`?

Reply:
Yes, `multierror` has a built in formatter that includes how many errors have occurred.